sendgrid/sendgrid.py

```diff
@@ -58,3 +57,14 @@ def api_key(self, value):
     @property
     def impersonate_subuser(self):
         return self._impersonate_subuser
+
+
+class SendGridClient(SendGridAPIClient):
+    def __init__(self, **opts):
+        super().__init__(**opts)
+
+    def send_mail(self, msg):
+        try:
+            return self.client.mail.send.post(request_body=msg.get())
+        except Exception as e:

```

Comment:
catching a urllib error here is still not ideal because it requires this library to know how the underlying client library was implemented. Thats why I've added this issue as well:
https://github.com/sendgrid/python-http-client/issues/16

Even here you are still not catching the HTTPException explicitly. How are you differentiating say a KeyError or an AttributeError versus an HTTPError?

Furthermore, the SendGridException as implemented is of little use here. A structured exception (similar to how HTTPError is defined) is more practical.

for example, I've defined these in my own wrapper.

```
class SendGridAPIError(Exception):
    def __init__(self, url, code, headers, body):
        self.url = url
        self.status_code = code
        self.headers = headers
        self.body = body

    def __unicode__(self):
        return u'{} [{}] {}'.format(self.url, self.status_code, self.body)  # pragma: no cover


class SendGridAPIRateLimitError(SendGridAPIError):
    pass
```

What this allows for is specific handling of errors and a way to inspect the error objects for specific details.